### PR TITLE
Fix LLD / LLM ts and eslint warnings

### PR DESCRIPTION
**Pull request by valpinkman**
- chore: update typescript eslint plugin + fix new errors
- chore: fix lint warnings in LLM
- chore: fix lint warnings in LLD

### 📝 Description

Update typscript eslint plugin to last version (support for TS 5) + fix all new errors
Fix LLM and LLD remaining eslint warnings
Add `tests` folder to typecheck in LLD + fixes errors and warnings

### ❓ Context

- **Impacted projects**: `live`, `lld`, `llm`, `ci` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Comment by cng-ledger** [comment]
@valpinkman 
I'm too new to Ledger to do a quality review on this. However just a suggestion, there is a `--max-warnings {Integer}` flag you can use with `eslint` which can cause CI workflow to fail if there are more than `x` amount of eslint warnings. 

This can enforce contributor to fix `eslint` warnings before merging the PR.

File: apps/ledger-live-mobile/src/families/ethereum/EthereumCustomFees/Ethereum1559CustomFees.tsx [hunk 1]
```diff
@@ -131,7 +131,9 @@ const Ethereum1559CustomFees = ({
   useEffect(() => {
     if (!transaction) return;
     setTransaction(bridge.updateTransaction(transaction, transactionPatch));
-  }, [maxPriorityFeePerGas, maxFeePerGas, transactionPatch]);
+    // TODO: Some of the dependencies were ommitted, is that by choice ?
```
Comment (haammar-ledger) [inline apps/ledger-live-mobile/src/families/ethereum/EthereumCustomFees/Ethereum1559CustomFees.tsx:134]: It likely was, I would avoid changing the behaviour and put a TS ignore instead

**Comment by valpinkman** [comment]
> @valpinkman I'm too new to Ledger to do a quality review on this. However just a suggestion, there is a `--max-warnings {Integer}` flag you can use with `eslint` which can cause CI workflow to fail if there are more than `x` amount of `eslint` warnings.
> 
> This can enforce contributor to fix `eslint` warnings before merging the PR.

@cng-ledger noted but we don't want to fail on warnings for now I believe.

File: apps/ledger-live-mobile/src/families/ethereum/EthereumCustomFees/Ethereum1559CustomFees.tsx (hunk 1 above)
Comment (valpinkman) [reply apps/ledger-live-mobile/src/families/ethereum/EthereumCustomFees/Ethereum1559CustomFees.tsx:134]: Noted, will fix

**Review by haammar-ledger — APPROVED** [review]
All good for blockchain support scope

**Review by cgrellard-ledger — APPROVED** [review]

**Pull request merged**